### PR TITLE
Merge `0.23` -> `main`

### PR DESCRIPTION
**Pull request by danicheg**

### CODE REVIEW

File: docs/versions.md [hunk 1]
```diff
@@ -25,7 +25,7 @@
 | ------------------------- | --------------------------------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ----- | ------ | ---- |
 | [@VERSION_1_0@](/v1/)     | @:style(version-label, dev)Milestone@:@ | @:icon(error) | @:icon(check) | @:icon(error) | 3.1           | 1.8           | 2.x   | 3.x    | 1.8+ |
 | [@VERSION_0_23@](/v0.23/) | @:style(version-label, stable)Stable@:@ | @:icon(error) | @:icon(check) | @:icon(check) | 3.1           | 1.8           | 2.x   | 3.x    | 1.8+ |
-| [@VERSION_0_22@](/v0.22/) | @:style(version-label, stable)Stable@:@ | @:icon(error) | @:icon(check) | @:icon(check) | 3.0           | @:icon(error) | 2.x   | 2.x    | 1.8+ |
+| [@VERSION_0_22@](/v0.22/) | @:style(version-label, eol)EOL@:@       | @:icon(error) | @:icon(check) | @:icon(check) | 3.0           | @:icon(error) | 2.x   | 2.x    | 1.8+ |
```
Comment (armanbilge) [inline docs/versions.md:28]: This one will go live as soon as we merge this PR. Should we release 0.22.13 first?

Comment (danicheg) [reply docs/versions.md:28]: I think our 'deprecation' is just about a declaration of intent to stop actively supporting the 0.22 series. And even before releasing the 0.22.13, we already decided to stop its maintenance (Scala Steward has stopped its work). So I don't see evil to have 0.22 declared EOL in docs right now.

Comment (armanbilge) [reply docs/versions.md:28]: Yes, I agree with that. I worried if a contributor will be concerned that their PR will never be released, but I think the only one is https://github.com/http4s/http4s/pull/6138 and we discussed releasing it in https://github.com/http4s/http4s/pull/6181#issuecomment-1079186817.

**Review by armanbilge — APPROVED** [review]
Thank you!

**Pull request merged**